nextjs-template/src/services/prisma-datastore/schema.prisma

```diff
@@ -12,33 +12,72 @@ model Project {
   createdAt  DateTime  @default(now()) @map("created_at")
   projectUID String    @map("project_uid") @unique
   name       String    @db.VarChar(50)
+  fleets     Fleet[]
   devices    Device[]
 
   @@map("project")
 }
 
+model Fleet {
+  id          Int      @id @default(autoincrement())
+
+// unique fleet uid in notehub
+  fleetUID    String   @unique
+
+  // the project the fleet belongs to
+  project     Project   @relation(fields: [projectUID], references: [projectUID])
+  projectUID  String
+
+  // devices assigned to fleet (implicit many-to-many relationship)
+  devices     Device[]
+
+  @@map("fleet")
+}
+
 model Device {
-  id                Int @id @default(autoincrement())
+  id            Int         @id @default(autoincrement())
 
   // unique device uid in notehub
-  deviceUID         String @unique @map("device_uid") @db.VarChar(64)
+  deviceUID     String      @unique @map("device_uid") @db.VarChar(64)
 
   // device serial number (cached)
-  name              String? @db.VarChar(80) // cached
-  locationName      String? @db.VarChar(200) // cached
-  lastSeenAt        DateTime? @map("last_seen_at") // the last time an event was heard from this device
+  name          String?     @db.VarChar(80) // cached
+  locationName  String?     @db.VarChar(200) // cached
+  lastSeenAt    DateTime?   @map("last_seen_at") // the last time an event was heard from this device
 
   // the project the device belongs to
-  project           Project @relation(fields: [project_id], references: [id])
-  project_id        Int
+  project       Project     @relation(fields: [projectUID], references: [projectUID])
+  projectUID    String
+
+  // fleets a device belongs to (implicit many-to-many relationship)
+  fleets        Fleet[]
```

Comment:
Will it be a problem storing the fleets a device is on considering users can change this information in Notehub, and we get notification of that update here?

Reply:
It shouldn't be. Each time a new event is routed into the app, I go out and fetch the fleets associated with that event's device from the Notehub API, so it will always be up to date. Intermittently, there does seem to be a `notify.qo` event that gets routed with the message `environment variable update received`, but I'm not confident enough in it to rely on that completely before fetching fresh fleet data from Notehub.

 In an ideal world, `fleetUIDs` would just be supplied with every event routed out of Notehub like we get for location, deviceID, etc., etc.